src/passes/OptimizeInstructions.cpp

```diff
@@ -2341,11 +2330,19 @@ struct OptimizeInstructions
           // exactness.
           if (ref == curr->ref && !needsExactCast) {
             if (needsNullCheck) {
-              replaceCurrent(builder.makeRefAs(RefAsNonNull, curr->ref));
+              ref = builder.makeRefAs(RefAsNonNull, ref);
+            }
+            if (curr->desc) {
+              // We must move the ref past the descriptor operand.
+              Index temp = builder.addVar(getFunction(), ref->type);
+              replaceCurrent(
+                builder.blockify(builder.makeLocalSet(temp, ref),
+                                 builder.makeDrop(curr->desc),
+                                 builder.makeLocalGet(temp, ref->type)));
```

Comment:
Another option is
```cpp
auto* block = ChildLocalizer(curr).getChildrenReplacement();
block->push_back(curr->ref);
```
It is safe to grab the ref like that, after we localize children, since then children have no effects.

This may be a little shorter, and also would avoid creating a local unnecessarily.

Reply:
That's fancy!